### ISSUE TITLE
FTP Installation Fatal Error

### ISSUE THREAD

**Issue by Regnalf** [kevinpapst/kimai2#494]
I've  tried the FTP installation process because i have only a shared server.

Went through all the normal installation process, not dev!
I used mysql not sqlite, worked so far (databases and user are created).
Pointed the domain entry point to /public/ and got this error:

Warning: require(): open_basedir restriction in effect. File(/data/web/e24498/html/kimai2-gg6/public../../vendor/autoload.php) is not within the allowed path(s): (/data/web/e24498/html/kimai2-gg6/public:/data/web/e24498/phpsession:/data/web/e24498/phptmp:/tmp:/usr/bin) in /data/web/e24498/html/kimai2-gg6/public/index.php on line 9

Warning: require(/data/web/e24498/html/kimai2-gg6/public../../vendor/autoload.php): failed to open stream: Operation not permitted in /data/web/e24498/html/kimai2-gg6/public/index.php on line 9

Warning: require(): open_basedir restriction in effect. File(/data/web/e24498/html/kimai2-gg6/public../../vendor/autoload.php) is not within the allowed path(s): (/data/web/e24498/html/kimai2-gg6/public:/data/web/e24498/phpsession:/data/web/e24498/phptmp:/tmp:/usr/bin) in /data/web/e24498/html/kimai2-gg6/public/index.php on line 9

Warning: require(/data/web/e24498/html/kimai2-gg6/public../../vendor/autoload.php): failed to open stream: Operation not permitted in /data/web/e24498/html/kimai2-gg6/public/index.php on line 9

Fatal error: require(): Failed opening required '/data/web/e24498/html/kimai2-gg6/public../../vendor/autoload.php' (include_path='.') in /data/web/e24498/html/kimai2-gg6/public/index.php on line 9

**Comment by kevinpapst** [comment]
Do you have the file `vendor/autoload.php`? 
Did you execute the `composer install` step?

**Comment by Regnalf** [comment]
I did the "composer install" step and i do have the "autoload.php" file under /vendor

**Comment by kevinpapst** [comment]
Oh and also: your poster seems to use the php setting `open_basedir` but limits access to files inside the document root, which is wrong. It seems that the allowed paths are:
`/data/web/e24498/html/kimai2-gg6/public:/data/web/e24498/phpsession:/data/web/e24498/phptmp:/tmp:/usr/bin`
but from your explanations I guess that your "shared root directory" is `/data/web/e24498/` or `/data/web/e24498/html/` but both are is not allowed, so Kimai cannot access the autoload file.

To sum it up: I believe your hoster has a fucked up PHP setup...

**Comment by Regnalf** [comment]
I think i found the solution for myself. I've checked the settings on the host-server and there was a php option "open_basedir" witch is standard set on active! 
Now i reached the login screen.
But does this mean that a script can access some other folders below /kimai2-gg6/ now?

**Comment by Regnalf** [comment]
Ok, i reached the login screen, but i couldn't login :-(
I tried it several times, even with a new generated user via terminal.
The users are in the database but i couldn't login. No message no error, password reset and register new user didn't work either (CSRF-Token issue)

**Comment by kevinpapst** [comment]
CSRF sounds like a session problem. Anything inside the error logs (both PHP/Server and Kimai at var/logs/production.log are important)

**Comment by Regnalf** [comment]
The "production.log" is not present, and the permission for folder /var is already 777.
In the php error log is only the fatal error from base_dir, but this is solved.

**Comment by kevinpapst** [comment]
And the permissions for the var/log and var/sessions directory?

**Comment by Regnalf** [comment]
All subdirectories in /var are set to 777

**Comment by kevinpapst** [comment]
Any file inside the sessions/prod directory after you submitted the login form?

**Comment by Regnalf** [comment]
No, but i found this in the error log, it's a warning
```
[11-Jan-2019 15:15:59 UTC] PHP Warning:  "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"? in /data/web/e24498/html/kimai2-gg6/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php on line 2718
[11-Jan-2019 15:15:59 UTC] PHP Warning:  "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"? in /data/web/e24498/html/kimai2-gg6/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php on line 2747

```

**Comment by kevinpapst** [comment]
If you have SSH, you should have SCP as well to do a proper installation.
I cannot support FTP as every hoster has different problems.
All the stuff you are experiencing is not related to Kimai, but to your environment.

**Comment by kevinpapst** [comment]
If you have ssh, you can simply install composer with wget or a similar command.

**Comment by kevinpapst** [comment]
No further feedback, closing

**Comment by kevinpapst** [comment]
Please see this issue #642 which adds a FTP package. 
There is a package for the upcoming version 0.9 linked which can be used for FTP installation.
